framework/src/main/java/org/checkerframework/common/value/PropertyFileHandler.java

```diff
@@ -0,0 +1,331 @@
+            return comparePropFileElementValue(subAnno, superAnno);
+        } else {
+            throw new BugInCF("We should never reach here.");
+        }
+    }
+
+    /**
+     * Try to read the value of the key in the provided property file.
+     *
+     * @param propFile the property file to open
+     * @param key the key to find in the property file
+     * @param defaultValue the default value of that key
+     * @return the value of the key in the property file
+     */
+    protected String readValueFromPropertyFile(String propFile, String key, String defaultValue) {
+        String res = null;
+        try {
+            Properties prop = new Properties();
+            ClassLoader cl = this.getClass().getClassLoader();
+            if (cl == null) {
+                // the class loader is null if the system class loader was
+                // used
+                cl = ClassLoader.getSystemClassLoader();
+            }
+            InputStream in = cl.getResourceAsStream(propFile);
+            if (in == null) {
+                // if the classloader didn't manage to load the file, try
+                // whether a FileInputStream works. For absolute paths this
+                // might help.
+                try {
+                    in = new FileInputStream(propFile);
+                } catch (FileNotFoundException e) {
+                    // ignore
+                }
+            }
+            if (in == null) {
+                checker.message(Kind.WARNING, "Couldn't find the properties file: " + propFile);
+                return null;
+            }
+            prop.load(in);
+            if (defaultValue == null) {
+                res = prop.getProperty(key);
+            } else {
+                res = prop.getProperty(key, defaultValue);
+            }
+        } catch (Exception e) {
+            checker.message(
+                    Kind.WARNING, "Exception in PropertyFileHandler.readPropertyFromFile: " + e);
+            e.printStackTrace();
+        }
+        return res;
+    }
+
+    /**
+     * Create a {@literal @}PropertyFile(value) annotation.
+     *
+     * @param value the value to set in the created annotation
+     * @return the created annotation
+     */
+    protected AnnotationMirror createPropertyFileAnnotation(String value) {
+        AnnotationBuilder builder = new AnnotationBuilder(env, PropertyFile.class);
+        builder.setValue("value", value);
+        return builder.build();
+    }
+
+    /**
+     * Create a {@literal @}StringVal(value) annotation.
+     *
+     * @param value the value to set in the created annotation
+     * @return the created annotation
+     */
+    protected AnnotationMirror createStringAnnotation(String value) {
+        return factory.createStringAnnotation(Collections.singletonList(value));
+    }
+
+    /**
+     * Get the first value of the {@literal @}StringVal() annotation.
```

Comment:
Why only first? Error if more than one?

Reply:
I decide to return null if there are more than one.